### PR TITLE
Extended tile inspector

### PR DESCRIPTION
**Pull request by Broxzier**
_This is still work-in-progress_

### Make sure to compile the sprite list before testing
To do so run: `openrct2.exe sprite build data\g2.dat ..\resources\g2`

Example

The biggest change to the tile inspector are the two new group boxes. When the player selects any of the elements, a group box with info, and another group box with properties will be shown. The content of those depends on the type. For rides it lists the ride name and track piece ID for example.

### Info
- [x] Terrain style
- [x] Edge style
- [x] Ownership status
- [x] Water level

### Settings
- [x] Remove park fences
- [x] Restore park fences
- [x] Raised corners
- [x] Raise / Lower

### Info
- [x] Name
- [x] Additions

### Settings
- [x] Raise / Lower
- [x] Change connected edges and corners

### Info
- [x] Ride type
- [x] Ride ID
- [x] Ride Name
- [x] Track piece ID
- [x] Track piece sequence

### Settings
- [x] Raise / Lower
- [x] Toggle lift hill
- [x] 'Apply-all' Raise / Lower
- [x] 'Apply-all' Toggle lift hill

### Info
- [x] Age
- [x] Quarter details

### Settings
- [x] Raise / Lower
- [x] Change quarter-tile location
- [x] Change quarter-collision flags

### Info
- [x] Type
- [x] Entrance part (0-2)
- [x] Station number (needs to be solved for park entrance left/right side)
- [x] Ride ID

### Settings
- [x] Raise / Lower

### Info
- [x] Type
- [x] Banner info

### Settings
- [x] Switch between flat, slanted right, and slanted left
- [x] Raise / Lower

### Info
- [x] Type
- [x] Part ID
- [x] Banner info

### Settings
- [x] Raise / Lower
- [ ] 'Apply-all' Raise / Lower

### Info
- [x] Banner info
- [x] No-Entry edges

### Settings
- [x] Raise / Lower

### Info
...

### Settings
- [x] Raise / Lower
- [x] Clamp to next element

## Other things
- [x] Scale-able interface and defines to easily extend it
- [x] Sort button, to sort elements on a tile based on base height
- [x] Spinners for changing the current X and Y
- [x] Rotating banners
- [x] Rotating flat paths (edges and corners)
- [ ] White selection area for both hovered and selected tile
- [x] Clipped strings for long entries
- [ ] Auto enable / disable buttons (e.g. disable 'Lower' when object is at base height 0)
- [ ] Different minimal height for each page
- [x] New corrupt elements spawn under the currently selected element

### CODE REVIEW

**Comment by marijnvdwerf** [comment]
'Track piece ID' should probably be 'Track type', right?

Also, I think it would be better to move the 'change ride type' stuff to the tile inspector (currently in the ride window), but I don't really have an idea as to how that could look.

**Comment by Gymnasiast** [comment]
@marijnvdwerf:
> Also, I think it would be better to move the 'change ride type' stuff to the tile inspector
 I disagree. It's something that concerns the entire ride (in about the same way as the chosen vehicle) and not a track element on a tile.

**Comment by IntelOrca** [comment]
@Gymnasiast until we implement the ability to swap ride type (track style) throughout the ride. But otherwise I agree.

**Comment by Broxzier** [comment]
> @marijnvdwerf 
> 'Track piece ID' should probably be 'Track type', right?

That's indeed what it means. At the time I made it I thought calling it type would be confusing.

File: src/localisation/string_ids.h [hunk 1]
```diff
@@ -1165,6 +1165,8 @@ enum {
 	STR_CONSTRUCTION_RIGHTS_NOT_FOR_SALE = 1727,
 	STR_CANT_BUY_CONSTRUCTION_RIGHTS_HERE = 1728,
 	STR_LAND_NOT_OWNED_BY_PARK = 1729,
+	STR_LAND_NOT_OWNED_AND_NOT_AVAILABLE = 5933,
```
Comment (Gymnasiast) [inline src/localisation/string_ids.h]: String IDs should be sorted numerically.

Comment (Broxzier) [inline src/localisation/string_ids.h]: I placed it there because they are very related, but I've moved everything down now.

**Comment by Broxzier** [comment]
The tile inspector is ready to be tested. I will rebase this on the develop branch tomorrow.

**Comment by janisozaur** [comment]
Remember to update g2 if you want to test it

**Comment by janisozaur** [comment]
Changes to `src/drawing/sprite.c` look superfluous.

**Comment by janisozaur** [comment]
https://github.com/Broxzier/OpenRCT2/blob/abccf603252fb3d9de3e354211a7ef7afb0415aa/src/windows/tile_inspector.c#L725 unused variable

https://github.com/Broxzier/OpenRCT2/blob/abccf603252fb3d9de3e354211a7ef7afb0415aa/src/windows/tile_inspector.c#L730 pointless, you override this value further down: https://github.com/Broxzier/OpenRCT2/blob/abccf603252fb3d9de3e354211a7ef7afb0415aa/src/windows/tile_inspector.c#L735

Same things in `window_tile_inspector_track_block_set_lift`

File: src/windows/guest_list.c [hunk 2]
```diff
@@ -200,7 +200,7 @@ void window_guest_list_open()
 	window->enabled_widgets =
 		(1 << WIDX_CLOSE) |
 		(1 << WIDX_PAGE_DROPDOWN) |
-		(1 << WIDX_PAGE_DROPDOWN_BUTTON) |
+		//(1 << WIDX_PAGE_DROPDOWN_BUTTON) |
```
Comment (Broxzier) [inline src/windows/guest_list.c]: Just noticed aswell. 65866365e333f530eb2693018409b33dd06612e5 is a messed up commit. will fix this today.

**Comment by janisozaur** [comment]
you check for `NULL` here: https://github.com/Broxzier/OpenRCT2/blob/abccf603252fb3d9de3e354211a7ef7afb0415aa/src/windows/tile_inspector.c#L680 but dereference them anyway: https://github.com/Broxzier/OpenRCT2/blob/abccf603252fb3d9de3e354211a7ef7afb0415aa/src/windows/tile_inspector.c#L686

**Comment by Broxzier** [comment]
@janisozaur: That check is to see if both pointers have been set, just to break out of the loop earlier. It can be replaced with `i > max(first, second)`, or even removed entirely (but making it continue iterate over the remaining elements).

`*firstElement = *secondElement;` copyies the second map element to the first element, swapping them.

**Comment by Broxzier** [comment]
Rebased!

File: data/language/en-GB.txt [hunk 3]
```diff
@@ -4230,8 +4230,70 @@ STR_5918    :{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{COM
 STR_5919    :{COMMA16}
 STR_5920    :Render weather effects
 STR_5921    :{SMALLFONT}{BLACK}If enabled, rain and gloomy colours will be rendered during storms.
+<<<<<<< HEAD
```
Comment (janisozaur) [inline data/language/en-GB.txt]: Shouldn't be here

Comment (Broxzier) [inline data/language/en-GB.txt]: Removed with a second rebase now.

**Comment by janisozaur** [comment]
Line 1335: stray `printf`

**Comment by janisozaur** [comment]
I have tested it a little now and posted my comments above. Otherwise it looks to do what it is meant to. 👍

**Comment by Broxzier** [comment]
@janisozaur Thanks for your suggestions, I'm working on it now. I did notice a very similar issue to the one you mentioned earlier today, but with a different button staying in pressed state. Reopening the window 'fixed' it, so there's probably something not initialized properly. I'll have to look into this.

**Comment by janisozaur** [comment]
@Broxzier can you rebase? #4490 introduced some conflicts.

**Comment by Broxzier** [comment]
@janisozaur: Sure, I hope this will be the last time before being merged... If anyone else still wants to test it, please do so ASAP.

**Comment by janisozaur** [comment]
I hope so as well. Lately Ted and Duncan have been busy with finishing off ride drawing, so it's understandable other PRs were on hold. I've tried helping them out by testing and reviewing, I believe it looks good, but I leave final decision to them.

**Comment by Broxzier** [comment]
@IntelOrca @duncanspumpkin If one of you could test and merge this, that would be appreciated. Rebasing on future conflicts will only costs me more time.

**Comment by IntelOrca** [comment]
@Broxzier two things which I will let you do in a follow up PR:

* collision is spelt wrong
* inspecting an entrance tile doesn't show base height correctly, just a blank box.

File: data/language/en-GB.txt [hunk 4]
```diff
@@ -4232,6 +4232,67 @@ STR_5920    :Render weather effects
...
 STR_5923    :{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{SMALLFONT}{BLACK}Max {COMMA16} {STRINGID} per train
+STR_5924    :Surface details
+STR_5925    :Path details
+STR_5926    :Track details
+STR_5927    :Scenery details
+STR_5928    :Entrance details
+STR_5929    :Fence details
+STR_5930    :Large scenery details
+STR_5931    :Banner details
+STR_5932    :Currupt element details
```
Comment (ThePsionic) [inline data/language/en-GB.txt]: Typo - `Currupt` should be `Corrupt`

**Pull request merged**